app/src/ui/preferences/appearance.tsx

```diff
@@ -21,7 +30,18 @@ const themes: ReadonlyArray<ISegmentedItem> = [
   },
 ]
 
-export class Appearance extends React.Component<IAppearanceProps, {}> {
+export class Appearance extends React.Component<
+  IAppearanceProps,
+  IAppearanceState
+> {
+  public constructor(props: IAppearanceProps) {
+    super(props)
+
+    this.state = {
+      automaticallySwitchTheme: this.props.automaticallySwitchTheme,
```

Comment:
💭 I'm curious if we actually need to track state here, given we don't for theme changes. I need to have a read of how stateful we're making other Preferences components, because it's not clear to me which stuff should be saved versus immediately updated...

Reply:
@shiftkey, from what I can tell State is needed as this.props is readonly :/

Reply:
I believe if you're updating `IAppState.automaticallySwitchTheme` through the dispatcher, and then emitting an update, that will trigger a re-render of the `Appearance` component with the updated prop value. That's why we mark the fields in `props` as `readonly` (because components shouldn't be updating them).

Reply:
@say25 all good, feel free to bump this PR when you get around to it